src/builder-utils.c

```diff
@@ -1615,6 +1615,95 @@ builder_maybe_host_spawnv (GFile                *dir,
   return flatpak_spawnv (dir, output, 0, error, argv);
 }
 
+int
+builder_get_all_checksums (const char *checksums[BUILDER_CHECKSUMS_LEN],
+                           GChecksumType checksums_type[BUILDER_CHECKSUMS_LEN],
+                           const char *md5,
+                           const char *sha1,
+                           const char *sha256,
+                           const char *sha512)
+{
+  int i = 0;
+
+  /* The order here is such that the first one is the
+     one to use by default if using a single one. That is
+     the longest checksum except it defaults to sha256
+     which was historically the one flatpak-builder used */
+
+  if (sha256 != NULL && *sha256 != 0)
+    {
+      checksums[i] = sha256;
+      checksums_type[i] = G_CHECKSUM_SHA256;
+      i++;
```

Comment:
It would be good to tie the definition of `BUILDER_CHECKSUMS_LEN` to this function, so that they can’t be adjusted out of step.

`g_assert (i < BUILDER_CHECKSUMS_LEN)`?